deployments/cloudfoundry/buildpack/bin/supply

```diff
@@ -0,0 +1,68 @@
+#!/bin/bash
+
+set -euo pipefail
+
+BUILD_DIR=$1
+CACHE_DIR=$2
+DEPS_DIR=$3
+IDX=$4
+
+BUILDPACK_DIR="$(readlink -f ${BASH_SOURCE%/*})"
+TARGET_DIR="$BUILD_DIR/.otelcollector"
+
+OTEL_CONFIG="${OTEL_CONFIG:-otelconfig.yaml}"
+# Version must be >=0.47.0, the first release included the enabled Cloud Foundry receiver
+OTEL_VERSION="${OTEL_VERSION:-0.47.0}"
+# Cloud Foundry only supports Linux stacks currently, but if that changes this variable can be modified
+# for the proper stack.
+OS="${OS:-linux_amd64}"
+
+echo "-----> Installing Otel Collector ${OTEL_VERSION}"
+echo "       BUILD_DIR: $BUILD_DIR"
+echo "       CACHE_DIR: $CACHE_DIR"
+echo "       DEPS_DIR: $DEPS_DIR"
+echo "       BUILDPACK_INDEX: $IDX"
+echo "       BUILDPACK_DIR: $BUILDPACK_DIR"
+echo "       TARGET_DIR: $TARGET_DIR"
+
+mkdir -p $TARGET_DIR
+
+# Copy over default OpenTelemetry configuration file for sidecar configuration use
+cp "$BUILDPACK_DIR/../$OTEL_CONFIG" "$TARGET_DIR/"
+
+# File name for given release version
+OTEL_BINARY="${OTEL_BINARY:-otelcontribcol_${OS}-v${OTEL_VERSION}}"
+OTEL_BINARY_TAR="$OTEL_BINARY.tar.gz"
+CACHED_OTEL_BINARY_TAR="$CACHE_DIR/$OTEL_BINARY_TAR"
+
+if [[ -f "$CACHED_OTEL_BINARY_TAR" ]]; then
+  echo "-----> Using cached Otel Collector install: $CACHED_OTEL_BINARY_TAR"
+  cp $CACHED_OTEL_BINARY_TAR $TARGET_DIR
+else
+  OTEL_BINARY_DOWNLOAD_URL="${OTEL_BINARY_DOWNLOAD_URL:-https://github.com/open-telemetry/opentelemetry-collector-releases/releases/download/v${OTEL_VERSION}/otelcol-contrib_${OTEL_VERSION}_${OS}.tar.gz}"
+  echo "-----> Downloading OpenTelemetry Collector v$OTEL_VERSION ($OTEL_BINARY_DOWNLOAD_URL)"
+  wget -nv -O "$TARGET_DIR/$OTEL_BINARY_TAR" $OTEL_BINARY_DOWNLOAD_URL
+fi
+
+if [ $? -ne 0 ]; then
```

Comment:
Since you have the `set -e` option at the beginning, the script will exit immediately before reaching this line if `wget` or any command before this line fails, and the message below will not be displayed.